job_bundles/README.md

```diff
@@ -0,0 +1,277 @@
+
+### Elements - Job Template
+
+The `template.json`/`template.yaml` file defines the runtime environment and all the processes that will run
+as part of an Amazon Deadline Cloud Job. It can be parameterized so that the same template can be used to
+create Jobs that differ only in their input values; much like a function or template in your favourite programming
+langauge. 
+
+For example, the [Job Template for the `blender_render` sample](https://github.com/aws-deadline/deadline-cloud-samples/blob/mainline/job_bundles/blender_render/template.yaml)
+defines input parameters like `BlenderSceneFile` which is a file path:
+
+```yaml
+- name: BlenderSceneFile
+  type: PATH
+  objectType: FILE
+  dataFlow: IN
+  userInterface:
+    control: CHOOSE_INPUT_FILE
+    label: Blender Scene File
+    groupLabel: Render Parameters
+    fileFilters:
+    - label: Blender Scene Files
+      patterns: ["*.blend"]
+    - label: All Files
+      patterns: ["*"]
+  description: >
+    Choose the Blender scene file you want to render. Use the 'Job Attachments' tab
+    to add textures and other files that the job needs.
+```
+
+The Job Parameter properties `userInterface`, `objectType`, and `dataFlow` are used by the Deadline Cloud CLI when
+present in a Job Template within a Job Bundle. 
+
+The `userInterface` properties control the behaviour of automatically generated Job submission UIs; both via 
+the `deadline bundle gui-submit` command line, and within Job submittion plugins for applications
+such as the [Autodesk Maya plugin](https://github.com/aws-deadline/deadline-cloud-for-maya). 
+In this example, the UI widget for inputting a value for `BlenderSceneFile` will be a file-selection dialog that
+allows filtering to see only Blender's `.blend` files or all files, and within a widget group called "Render Parameters":
+
+![BlenderSceneFile UI Widget](../.images/blender_submit_scene_file_widget.png)
+
+The `objectType` and `dataFlow` properties control the behaviour of Deadline Cloud's 
+[Job Attachments feature](https://docs.aws.amazon.com/deadline-cloud/latest/userguide/storage-job-attachments.html) when submitting
+a Job from the Job Bundle. In this case, `objectType: FILE` and `dataFlow: IN` mean that the value of `BlenderSceneFile`
+will be treated as an input file for Job Attachments. Contrast that with the definition of the `OutputDir` which has
+`objectType: DIRECTORY` and `dataFlow: OUT`:
+
+```yaml
+- name: OutputDir
+  type: PATH
+  objectType: DIRECTORY
+  dataFlow: OUT
+  userInterface:
+    control: CHOOSE_DIRECTORY
+    label: Output Directory
+    groupLabel: Render Parameters
+  default: "./output"
+  description: Choose the render output directory.
+```
+
+The value of `OutputDir` is treated by Job Attachments as a directory where the Job is expected to write output files.
+
+The remainder of the `blender_render` sample's Job Template defines the workflow as a single Step with each frame of the animation
```

Comment:
Perhaps a link out to OpenJD docs about this, or at least explaining the `embeddedFiles` perhaps?